File: back-end/src/main/java/com/joaoalencar/projetoimpacta/domain/model/supplier/Supplier.java
package com.joaoalencar.projetoimpacta.domain.model.supplier;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "suppliers")
@Getter
@Setter
public class Supplier {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "name")
    private String name;

    @Column(name = "document")
    private String document;
}

File: api/Domain/Entities/Supplier.cs
namespace api.Domain.Entities
{
    public class Supplier
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Document { get; set; }
        
        public Supplier()
        {
        }

        public Supplier(string name, string document)
        {
            Name = name;
            Document = document;
        }

        public ICollection<Product> Products { get; set; } = default!;
    }
}